### DISCUSSION TITLE
Spring Webflux Auto Instrumentation with additional tags

### DISCUSSION THREAD

**Discussion by algirdelizzz** [open-telemetry/opentelemetry-java-instrumentation#7696] [Q&A]
Hello,

I'm running a Spring Boot application along side the java-agent, it all works fine, but let's say I want to add some additional custom tags to the spans created by the io.opentelemetry.spring-webflux-5.0 library, what would be the correct approach of doing this? In manual mode, I guess it would be adding an additional attribute extractor in SpringWebfluxTelemetryBuilder.

**Comment by mateuszrzeszutek** [comment]
Hey @algirdelizzz ,

That depends whether you want to add attributes to the `SERVER` spans (incoming requests) or `CLIENT` spans (requests sent by your application).

* `SERVER` spans: the simplest way to do that would be to call [`Span.current()`](https://opentelemetry.io/docs/instrumentation/java/manual/#get-the-current-span) and set whatever attributes you want somewhere in your handler functions.
* `CLIENT` spans: you'd have to build a `SpringWebfluxTelemetry` instance and install it yourself; passing a custom `AttributesExtractor` to the builder. The agent will back off and use your own manually installed instrumentation then.
